controls/srg_ctr/SRG-APP-000290-CTR-000670.yml

```diff
@@ -6,18 +6,8 @@ controls:
     of audit tools.
   related:rules:
   - audit_log_forwarding_uses_tls
-  status: inherently met
-  status_justification: |-
-    The audit tools used with OpenShift Container Platform are all delivered as container images referenced by manifest checksum. Installation sources for OpenShift are also delivered as container images, referenced by manifest checksum. See, for example, https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable-4.8/release.txt
-
-    The listing of platform components by their manifest checksums, as well as the installer and command line tooling, are additionally checksummed. See, for example, https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable-4.8/sha256sum.txt
-
-    The file that contains the checksums of the installer tooling and listing of container image manifest hashes is signed and the signature, validatable with Red Hat's product security GPG key, is published alongside the listing. See, for example, https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable-4.8/sha256sum.txt.gpg
-
-    Included in the release of the base platform are the components that index the available installation sources for additional components, delivered as Operators, from Red Hat. The images for the OpenShift Logging Operator, which are the only supported mechanism for exporting audit logs from the cluster and forwarding to an external log aggregation solution, 
-  artifact_description: |-
-    Supporting evidence is in the following documentation
-
-    https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable-4.8/release.txt
-    https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable-4.8/sha256sum.txt
-    https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable-4.8/sha256sum.txt.gpg
+  rules:
+  - auditd_log_format
+  - auditd_data_disk_error_action
+  - auditd_data_retention_max_log_file_action_stig
+  status: automated
```

Comment:
yes and no. I think in general the tampering is best protected by offloading the audit logs off the cluster, which id done by the cluster-logging-operator and its configuration. In this case, we added the rules that you see to make sure that audit cannot be just flooded with extra messages (dist_error_action and mx_log_file_action), make sure that timestamps and other metadata is there (auditd_log_format).

Do you think based on the vulndiscission that this SRG is missing some rules?

Note that the draft in the spreasheet collapsed quite a few SRGs into a single row, so we might need to change the title and the vulndiscussion.

In case I'm not answering your question - feel free to just reframe the question or propose another fix.

Reply:
I see the point you are making, off loading logs help protect their integrity, but the approach seems a bit off topic.
For example, `SRG-APP-000111-CTR-000220` seems to cover sending of logs already: https://github.com/ComplianceAsCode/content/blob/master/controls/srg_ctr/SRG-APP-000111-CTR-000220.yml
But I'm not experienced with interpreting, or assigning rules to SRGs though.

Now, I have re-read  `000670`  from `shared/references/disa-ctr-srg-v1r3.xml` and I think the SRG is only about integrity of the tooling, so it should probably have rules that check that the audit binary, or that the logging operator was not tampered with.